GitCommands/PathUtil.cs

```diff
@@ -72,20 +72,37 @@ public static bool IsLocalFile(string fileName)
         }
 
         /// <summary>
-        /// A naive way to check whether the given path is a URL by checking
-        /// whether it starts with either 'http', 'ssh' or 'git'.
+        /// Check whether the given string contains one or more URLs and extract
+        /// any URLs that exist.
         /// </summary>
-        /// <param name="path">A path to check.</param>
-        /// <returns><see langword="true"/> if the given path starts with 'http', 'ssh' or 'git'; otherwise <see langword="false"/>.</returns>
-        [Pure]
-        public static bool IsUrl(string path)
+        /// <remarks>
+        /// Uri.IsWellFormedUriString is used instead of Uri.TryCreate as file URIs
+        /// of the form X:\\directory\\filename should not be treated as URLs.
+        /// Each URL extracted from <paramref name="contents"/> is added as an entry in
+        /// <paramref name="urls"/>.
+        /// </remarks>
+        /// <param name="contents">A string to attempt to extract URLs from.</param>
+        /// <param name="urls">A <see cref="List{T}"/> that contains any URLs extracted from <paramref name="contents"/>.</param>
+        /// <returns><see langword="true"/> if the given string contains one or more URLs; otherwise <see langword="false"/>.</returns>
+        public static bool TryExtractURLsFromString(string contents, out List<string> urls)
```

Comment:
💭  To clarify - we now have a single callsite for this method in `FormClone`, right? If this is so, I'd turn this method into a private in `FormClone`, and test it through a test accessor. And since the only use case for this method is to use the first detected URL, there are questions why we need to allocate a list, etc.

Reply:
Thank you for the feedback; I've changed this method to a private in `FormClone` and have added a test for it.